### PR TITLE
Added module for reverting fishing rod velocity back to 1.8

### PR DESCRIPTION
**Pull request by MWHunter**
The formulas were taken directly from 1.8, although due to other changes, I reduced the final multiplier from 1.5 to 1.37 to better resemble 1.8 rod distance.  Perhaps gravity and air friction are also calculated differently in 1.9+

This fixes issue #149

### CODE REVIEW

File: src/main/resources/config.yml [hunk 1]
```diff
@@ -246,6 +246,10 @@ old-fishing-knockback:
   # This is the delay in milliseconds in-between rod damage, so the player hit has time to fall back down
   hitCooldown: 1000
 
+old-fishing-rod-velocity:
```
Comment (kernitus) [inline src/main/resources/config.yml:249]: Please have a consistent naming scheme. The module should just be called `fishing-rod-velocity` in both the class and config.

File: src/main/resources/config.yml [hunk 2]
```diff
@@ -246,6 +246,10 @@ old-fishing-knockback:
...
+  # This is to revert the fishing rod cast to pre-1.9 velocity
```
Comment (kernitus) [inline src/main/resources/config.yml:250]: Please provide a little bit more context, such as that the fishing rod goes farther etc.

File: src/main/java/kernitus/plugin/OldCombatMechanics/module/ModuleFishingRodVelocity.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                Player casterPlayer = (Player) caster;
+                double playerYaw = casterPlayer.getLocation().getYaw();
+                double playerPitch = casterPlayer.getLocation().getPitch();
+
+                float oldMaxVelocity = 0.4F;
+                double velocityX = -Math.sin(playerYaw / 180.0F * (float)Math.PI) * Math.cos(playerPitch / 180.0F * (float)Math.PI) * oldMaxVelocity;
+                double velocityZ = Math.cos(playerYaw / 180.0F * (float)Math.PI) * Math.cos(playerPitch / 180.0F * (float)Math.PI) * oldMaxVelocity;
+                double velocityY = -Math.sin(playerPitch / 180.0F * (float)Math.PI) * oldMaxVelocity;
+
+                double oldVelocityMultiplier = 1.37F; // This is 1.5 in vanilla 1.8, the formula doesn't work exactly as it should
```
Comment (kernitus) [inline src/main/java/kernitus/plugin/OldCombatMechanics/module/ModuleFishingRodVelocity.java:43]: If this number is not how it was in 1.8, then is should probably be configurable in the config.yml.

**Review by kernitus — CHANGES_REQUESTED** [review]
Thank you for taking your time to implement this. I do not currently have the time to test this, however I have a couple of questions:

1. Where did you specifically find the original 1.8 formulae, and can you please provide such code
2. Why was it necessary to change the final multiplier, and how did you decide on the new number

**Comment by MWHunter** [comment]
1. The code was taken from Mod Coder's Pack 1.8.9 in EntityFishHook.java and is being compared against the official 1.16.3 mappings.  This research is my own.  I'll put psuedocode below because of legal issues with sharing mappings.
2. I dug deeper into the code and found that the gravity in 1.8 is 0.04 blocks per second while in 1.16 it is 0.03 blocks per second, for fishing rods.  1.8 fishing rods could be achieved correctly if every tick, 0.01 block per second of vertical velocity is added to the fishing rod if it is in the air and hasn't collided with anything and before friction is applied, either that or after friction adding 0.01 / 0.92 instead, to cancel out with the 0.92 air friction multiplier.  The old multiplier was calculated by getting the 1.8 and 1.16 rods to go the same horizontal distance when looking straight which kind of works.  When I have time, I will attempt this better and more accurate solution.  I will also have to look into differences between versions because even 1.9 and 1.16 aren't identical with fishing rod behavior!

1.8 psuedocode:
double1 = (blocks of water above us * 2) - 1
hook's y motion += 0.04 * double1
hook's y motion *= 0.92

1.16 psuedocode:
if we aren't in water subtract 0.03 from the hook's y motion
hook's y motion *= 0.92

**Comment by MWHunter** [comment]
This module has been tested on 1.9 to 1.16, and on all eight versions the rod now goes twelve blocks, just like in 1.8.  The module detects if it is on version 1.9-1.13 to check whether it needs to change the rod's velocity, because gravity did not change on these versions in testing, while on 1.14-1.16 gravity did change.

**Comment by kernitus** [comment]
Is there a fallback value in case the server version can't be detected?

Could we make the actual values configurable?

**Comment by MWHunter** [comment]
Initial rod velocity: 1.9 changes this value
Rod gravity: 1.14 changes this value

The code always changes initial rod velocity because, on all versions from 1.9 onward, it is not the 1.8 values.  The code checks for 1.9-1.13 server versions because then it doesn't need to change the rod's gravity, because it is already 1.8 values.  The code falls back to 1.14 values and only checks for the special 1.9-1.13 server versions.

I did not include a config because users aren't going to understand this and because the code achieves exactly 1.8 values.

**Comment by kernitus** [comment]
So, I did a bit of testing. I have no idea how you performed your calculations or testing, because on 1.16 with the module disabled the hook easily reaches 12 blocks away. If I enable this module, it then goes 16 blocks away. That doesn't seem to be the behaviour you'd described?

**Comment by MWHunter** [comment]
I cannot your behavior at all.  I am measuring looking directly at yaw 0.0 pitch 0.0.  On 1.16.4, on Craftbukkit, Spigot, and Paper, the rod goes 10 blocks without the module and 12 blocks with the module.  Could be a bug with another plugin or another server jar, but it works on my machine.

**Comment by kernitus** [comment]
Yes, you're right. I wasn't testing it at exactly 0° pitch.

**Comment by kernitus** [comment]
Alright - so after some extra testing, I can't exactly figure out what you're doing with the gravity. The hook seems to land close if the task that modifies the velocity is enabled. However, the 1.14+ gravity is greater, so surely we should be *adding* 0.01 to the Y value to counter that?

**Comment by MWHunter** [comment]
Gravity is less on 1.14+.  I made a bad comment in the code.  The comment should be

` * Fishing rod gravity in 1.14+ is 0.03 while in 1.8 it is 0.04`


Also the BukkitVersion variable is unused.

**Pull request merged**